### PR TITLE
Add federatedIdentityClientId

### PR DESCRIPTION
**Pull request by jackrichins**
Added property to support cross-tenant scenarios.

### CODE REVIEW

**Comment by jackrichins** [comment]
@pilor @HarishRane for review

**Review by pilor — APPROVED** [review]

**Comment by pilor** [comment]
@jackrichins let me know when you are ready for this to merge (i.e. after Harish approves)

File: v1.0/common-api-contracts.md [hunk 1]
```diff
@@ -162,6 +162,11 @@ For resources that implement data encryption and allow the customer to specify t
                                 "type": "string",
                                 "defaultValue": "",
                                 "description": "user assigned identity to use for accessing key encryption key Url. Ex: /subscriptions/fa5fc227-a624-475e-b696-cdd604c735bc/resourceGroups/<resource group>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myId. Mutually exclusive with identityType systemAssignedIdentity.”
+                            },
+                            "federatedIdentityClientId": {
```
Comment (HarishRane) [inline v1.0/common-api-contracts.md:166]: Per original communication, this is supposed to be named as "federatedClientId". Other 2 RPs are using that name and one RP seems to be using "federatedIdentityClientId".

Comment (jackrichins) [reply v1.0/common-api-contracts.md]: Updated to federatedClientId.

**Pull request merged**